lib/charms/grafana_cloud_integrator/v0/cloud_config_provider.py

```diff
@@ -49,9 +49,6 @@ def _on_relation_changed(self, event):
         if not self._charm.unit.is_leader():
             return
         
-        if not self._charm.credentials_configured:
-            return
-
```

Comment:
Is it safe to enter the loop below without this guard?

Reply:
Moved it in the for loop :)